components/authz-service/storage/v2/postgres/postgres_test.go

```diff
@@ -4614,8 +4617,175 @@ func TestDeleteRule(t *testing.T) {
 	}
 }
 
+func TestApplyStagedRules(t *testing.T) {
+	store, db, _, _ := testhelpers.SetupTestDB(t)
+	defer db.CloseDB(t)
+	defer store.Close()
+	// No project filter concerns in these tests so safe to re-use context.
+	ctx := context.Background()
+
+	cases := []struct {
+		desc string
+		f    func(*testing.T)
+	}{
+		{"when there are no staged rules, applied rules are unchanged", func(t *testing.T) {
+			projID := "project-1"
+			insertTestProject(t, db, projID, "let's go jigglypuff - topsecret", storage.Custom)
+			rule := insertAppliedRuleWithMultipleConditions(t, db, projID, storage.Node)
+			assertCount(t, 1, db.QueryRow(`SELECT count(*) FROM iam_project_rules
+				WHERE id=$1 AND project_id=$2 AND name=$3 AND type=$4`, rule.ID, rule.ProjectID, rule.Name, rule.Type.String()))
+
+			err := store.ApplyStagedRules(ctx)
+			assert.NoError(t, err)
+			assertCount(t, 1, db.QueryRow(`SELECT count(*) FROM iam_project_rules
+				WHERE id=$1 AND project_id=$2 AND name=$3 AND type=$4`, rule.ID, rule.ProjectID, rule.Name, rule.Type.String()))
+		}},
+		{"when there are n staged rules marked for update but no applied rules, it creates n staged rules", func(t *testing.T) {
```

Comment:
Test name does not sound right; should it end with "... it creates n applied rules" ?